diff -uNr python-rpi-ws281x-4.3.0/lib/rpihw.c python-rpi-ws281x-4.3.0.new/lib/rpihw.c
--- python-rpi-ws281x-4.3.0/lib/rpihw.c	2021-06-01 16:24:53.000000000 +0300
+++ python-rpi-ws281x-4.3.0.new/lib/rpihw.c	2022-06-28 21:24:40.996431868 +0300
@@ -159,6 +159,13 @@
         .videocore_base = VIDEOCORE_BASE_RPI2,
         .desc = "Compute Module 4 v1.0 WiFi 8GB"
     },
+    {
+        .hwver = 0xb03141,
+        .type = RPI_HWVER_TYPE_PI4,
+        .periph_base = PERIPH_BASE_RPI4,
+        .videocore_base = VIDEOCORE_BASE_RPI2,
+        .desc = "Compute Module 4 v1.1"
+    },
     //
     // Model B Rev 1.0
     //
